### PR TITLE
Feat/curriculum service todos

### PR DESCRIPTION
**Pull request by jdiester**

### CODE REVIEW

**Review by danieldietzler — APPROVED** [review]
LGTM

File: server/src/main/java/de/uftos/controller/ConstraintController.java [hunk 1]
```diff
@@ -27,6 +29,8 @@
 public class ConstraintController {
   private final ConstraintSignatureService constraintSignatureService;
   private final ConstraintInstanceService constraintInstanceService;
+  private final ConstraintInstanceRepository constraintInstanceRepository;
+  private final ConstraintSignatureRepository constraintSignatureRepository;
```
Comment (SuperMarcomen) [inline server/src/main/java/de/uftos/controller/ConstraintController.java:33]: Why is this needed?

File: server/src/main/java/de/uftos/dto/CurriculumRequestDto.java [hunk 2]
```diff
@@ -11,15 +12,18 @@
...
   /**
    * Maps the information from the data transfer object to a new curriculum entity.
    *
+   * @param grade the Grade object to be mapped
    * @return the new curriculum entity.
    */
-  public Curriculum map() {
-    return new Curriculum(this.gradeId, this.lessonsCounts);
+  public Curriculum map(Grade grade) {
+    return new Curriculum(grade, this.name, this.lessonsCounts);
```
Comment (SuperMarcomen) [inline server/src/main/java/de/uftos/dto/CurriculumRequestDto.java:26]: I think that you don't need the Grade instance and that
```java
return new Curriculum(new Grade(gradeId), this.name, this.lessonsCounts);
```


should work just fine

File: server/src/main/java/de/uftos/dto/CurriculumResponseDto.java [hunk 3]
```diff
@@ -16,4 +18,19 @@
...
+
+
+  /**
+   * Creates a CurriculumResponseDto from a Curriculum.
+   *
+   * @param curriculum the Curriculum whose information should be included
+   *                   in the CurriculumResponseDto.
+   * @return the CurriculumResponseDto containing information about the Curriculum.
+   */
+  public static CurriculumResponseDto createResponseDtoFromCurriculum(Curriculum curriculum) {
```
Comment (SuperMarcomen) [inline server/src/main/java/de/uftos/dto/CurriculumResponseDto.java:30]: Shouldn't this be a constructor?

**Review by SuperMarcomen — CHANGES_REQUESTED** [review]

**Review by SuperMarcomen — APPROVED** [review]
